### PR TITLE
MAP_FIXED fix

### PR DESCRIPTION
**Pull request by brooksdavis**
A fix for #1365

### CODE REVIEW

File: sys/vm/vm_map.c [hunk 1]
```diff
@@ -4363,10 +4363,10 @@ vm_map_delete(vm_map_t map, vm_offset_t start, vm_offset_t end,
 	 * Find the start of the region, and clip it.
 	 * Step through all entries in this region.
 	 */
-	rv = vm_map_lookup_clip_start(map, start, &entry, &scratch_entry);
+	rv = vm_map_lookup_clip_start(map, start, &entry, &prev_entry);
 	if (rv != KERN_SUCCESS)
 		return (rv);
-	for (; entry->start < end; entry = next_entry) {
+	for (; entry->start < end; prev_entry = entry, entry = next_entry) {
```
Comment (bsdjhb) [inline sys/vm/vm_map.c:4369]: So this is the real fix, updating prev_entry when iterating the loop?

**Comment by bsdjhb** [comment]
Earlier on the issue I think you had a patch that added assertions, do we want to keep any of those assertions?

File: sys/vm/vm_map.c (hunk 1 above)
Comment (brooksdavis) [reply sys/vm/vm_map.c:4369]: Yeah, updating `prev_entry` is the fix. The rest of renaming it from `scratch_entry` to `prev_entry`.

**Comment by brooksdavis** [comment]
> Earlier on the issue I think you had a patch that added assertions, do we want to keep any of those assertions?

In the end I didn't end up with any assertions that were both correct and fired for this bug. I think I could now construct one that does with a bit of rejiggering of the logic, but I'm not sure it's worth it.

**Comment by brooksdavis** [comment]
I've added some asserts under DIAGNOSTIC as suggested by @qwattash in the issue.

**Review by bsdjhb — APPROVED** [review]

**Review by nwf — APPROVED** [review]

**Pull request merged**